internal/credential/verification.go

```diff
@@ -0,0 +1,176 @@
+package credential
+
+import (
+	"crypto"
+	"fmt"
+
+	credsdk "github.com/TBD54566975/ssi-sdk/credential"
+	"github.com/TBD54566975/ssi-sdk/credential/signing"
+	"github.com/TBD54566975/ssi-sdk/credential/verification"
+	"github.com/TBD54566975/ssi-sdk/cryptosuite"
+	didsdk "github.com/TBD54566975/ssi-sdk/did"
+	"github.com/goccy/go-json"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/mr-tron/base58"
+	"github.com/multiformats/go-multibase"
+	"github.com/multiformats/go-varint"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/tbd54566975/ssi-service/internal/keyaccess"
+	"github.com/tbd54566975/ssi-service/internal/util"
+)
+
+type CredentialVerifier struct {
+	verifier *verification.CredentialVerifier
+	resolver *didsdk.Resolver
+}
+
+// NewCredentialVerifier creates a new credential verifier which executes both signature and static verification checks.
+// In the future the set of verification checks will be configurable.
+func NewCredentialVerifier(resolver *didsdk.Resolver) (*CredentialVerifier, error) {
+	if resolver == nil {
+		return nil, errors.New("resolver is nil")
+	}
+	// TODO(gabe): consider making this configurable
+	verifiers := verification.KnownVerifiers
+	verifier, err := verification.NewCredentialVerifier(verifiers)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create static credential verifier")
+	}
+	return &CredentialVerifier{
+		verifier: verifier,
+		resolver: resolver,
+	}, nil
+}
+
+func (v CredentialVerifier) VerifyJWTCredential(token string) error {
+	cred, err := signing.ParseVerifiableCredentialFromJWT(token)
+	if err != nil {
+		return util.LoggingErrorMsg(err, "could not parse credential from JWT")
+	}
+	kid, pubKey, err := v.getVerificationInformation(cred.Issuer.(string), "")
+	if err != nil {
+		return util.LoggingErrorMsg(err, "could not get verification from credential")
+	}
+	verifier, err := keyaccess.NewJWKKeyAccessVerifier(kid, pubKey)
+	if err != nil {
+		return util.LoggingErrorMsg(err, "could not create verifier")
+	}
+	if err := verifier.Verify(keyaccess.JWKToken{Token: token}); err != nil {
```

Comment:
Now I can see wrapping with a JWKToken{} object instead of just using a string may become a nucance